File: src/GFG/Array/Videos/MoveZerosToEnd.java
package GFG.Array.Videos;

public class MoveZerosToEnd {
    static void moveToEnd(int[] arr) {

        int count = 0;
        // for (int i = 0; i < arr.length; i++) {
        // if (arr[i] != 0) {
        // arr[count++] = arr[i];
        // }
        // }
        // for (int i = count; i < arr.length; i++) {
        // arr[i] = 0;
        // }
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == 0) {
                count++;
            } else if (count > 0) {
                arr[i - count] = arr[i];
                arr[i] = 0;
            }
        }
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }

    public static void main(String[] args) {
        int arr[] = { 0, 1, 9, 8, 4, 0, 0, 2, 7, 0, 6, 0, 9 };
        moveToEnd(arr);
    }
}

File: LeetCode/Arrays/Medium/MoveZerosToEnd.js
const nums = [0, 1, 0, 3, 0, 12];
const moveToEnd = (nums) => {
  let count = 0;
  for (let index = 0; index < nums.length; index++) {
    if (nums[index] !== 0) {
      nums[count++] = nums[index];
      nums[index] = 0;
    }
  }
  //   for (let i = 0; i < nums.legth; i++) {
  //     if (nums[i] == 0) {
  //       count++;
  //     } else if (count > 0) {
  //       nums[i - count] = nums[i];
  //       nums[i] = 0;
  //     }
  //   }

  return nums;
};
console.log(moveToEnd(nums));
